Source/CesiumRuntime/Private/CesiumTextureUtility.cpp

```diff
@@ -670,4 +762,93 @@ TextureAddress convertGltfWrapTToUnreal(int32_t wrapT) {
+    const CesiumGltf::Sampler& sampler,
+    CesiumGltf::ImageCesium& image) {
+  std::unique_lock lock(textureResourceMutex);
+
+  ExtensionUnrealTextureResource& extension =
+      image.addExtension<ExtensionUnrealTextureResource>();
+
+  // Future already exists, we don't need to do anything else.
+  if (extension.preprocessFuture.has_value()) {
+    return extension.preprocessFuture.value();
+  }
+
+  // Generate mipmaps if needed.
+  // An image needs mipmaps generated for it if:
+  // 1. It is used by a Texture that has a Sampler with a mipmap filtering
+  // mode, and
+  // 2. It does not already have mipmaps.
+  // It's ok if an image has mipmaps even if not all textures will use them.
+  // There's no reason to have two RHI textures, one with and one without
+  // mips.
+  bool needsMipmaps;
+  switch (sampler.minFilter.value_or(
+      CesiumGltf::Sampler::MinFilter::LINEAR_MIPMAP_LINEAR)) {
+  case CesiumGltf::Sampler::MinFilter::LINEAR_MIPMAP_LINEAR:
+  case CesiumGltf::Sampler::MinFilter::LINEAR_MIPMAP_NEAREST:
+  case CesiumGltf::Sampler::MinFilter::NEAREST_MIPMAP_LINEAR:
+  case CesiumGltf::Sampler::MinFilter::NEAREST_MIPMAP_NEAREST:
+    needsMipmaps = true;
+    break;
+  default: // LINEAR and NEAREST
+    needsMipmaps = false;
+    break;
+  }
+
+  if (!needsMipmaps || image.pixelData.empty()) {
+    // If we don't need mipmaps, we don't want to create a future for them.
+    // This allows a future sampler using this image that does need mipmaps to
+    // generate them.
+    return std::nullopt;
+  }
+
+  CesiumAsync::Promise<CesiumGltf::ImageCesium*> promise =
+      asyncSystem.createPromise<CesiumGltf::ImageCesium*>();
+
+  extension.preprocessFuture = promise.getFuture().share();
+
+  lock.unlock();
+
+  // We need mipmaps generated.
+  std::optional<std::string> errorMessage =
+      CesiumGltfReader::GltfReader::generateMipMaps(image);
+  if (errorMessage) {
+    UE_LOG(
+        LogCesium,
+        Warning,
+        TEXT("%s"),
+        UTF8_TO_TCHAR(errorMessage->c_str()));
+  }
+  promise.resolve(&image);
+  return *extension.preprocessFuture;
+}
+
+CesiumAsync::SharedFuture<void> createMipMapsForAllTextures(
+    const CesiumAsync::AsyncSystem& asyncSystem,
+    CesiumGltf::Model& model) {
+  std::vector<CesiumAsync::SharedFuture<CesiumGltf::ImageCesium*>> futures;
+  for (const Texture& texture : model.textures) {
+    const CesiumGltf::Sampler& sampler =
+        model.getSafe(model.samplers, texture.sampler);
+    std::optional<CesiumAsync::SharedFuture<CesiumGltf::ImageCesium*>>
+        optionalFuture = createMipMapsForSampler(
+            asyncSystem,
+            sampler,
+            *model.images[texture.source].cesium);
+    if (optionalFuture.has_value()) {
+      futures.push_back(optionalFuture.value());
+    }
+  }
+
+  return asyncSystem.all(std::move(futures))
+      .thenImmediately(
+          []([[maybe_unused]] std::vector<CesiumGltf::ImageCesium*>&& results)
```

Comment:
```suggestion
          [](std::vector<CesiumGltf::ImageCesium*>&& /*results*/)
```

From our style guide: 
https://github.com/CesiumGS/cesium-native/blob/main/doc/style-guide.md#%EF%B8%8F-other